### PR TITLE
Add gateway-api and refactor LB config

### PR DESCRIPTION
**Pull request by alvarorm22**
**Description of the change**

- [x] Add Gateway-api for GKE 
- [x] Refactor Load Balancer config and options
- [x] Support AWS LB 
- [x] Remove Ingress 
- [x] Support using custom certs and managed certs
- [x] Refactor char values for certificates, gateway and router 
- [x] Rotate custom certificates in case the cert content change 

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

### CODE REVIEW

**Comment by alvarorm22** [comment]
https://app.shortcut.com/cartoteam/story/383188/configure-auto-loadbalancer-in-aws

**Comment by alvarorm22** [comment]
https://app.shortcut.com/cartoteam/story/383840/gke-ingress-cert-rotation

**Comment by alvarorm22** [comment]
https://app.shortcut.com/cartoteam/story/383186/configure-automanaged-certs-on-gke

File: chart/templates/gateway/gcp-backend-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+# FIXME: Delete when gateway support tiemouts
+# https://gateway-api.sigs.k8s.io/geps/gep-1742/
```
Comment (moicalcob) [inline chart/templates/gateway/gcp-backend-policy.yaml:3]: Can we create a story for it so that we don't lose track of this one? 😄

Comment (alvarorm22) [reply chart/templates/gateway/gcp-backend-policy.yaml:3]: created 👍🏼

**Comment by alvarorm22** [comment]
https://app.shortcut.com/cartoteam/story/386619

**Review by moicalcob — DISMISSED** [review]
Nice job! 🙌🏼

**Review by moicalcob — APPROVED** [review]

**Pull request merged**